src/features/live_sensor/utils.ts

```diff
@@ -0,0 +1,90 @@
+import { notificationServiceInstance } from '~core/notificationServiceInstance';
+import { SENSOR_PRESICION } from './constants';
+import type { SensorDataAtomExportType } from './atoms/sensorData';
+import type { TriggerRequestActionType } from './atoms/sensorResource';
+
+export function hookSensors(
+  sensorDataAtom: SensorDataAtomExportType,
+  stopRecording: () => void,
+  accelerometer: Accelerometer,
+  orientationSensor: AbsoluteOrientationSensor,
+  gyroscope: Gyroscope,
+) {
+  // Describe accelerometer
+  accelerometer.onreading = () => {
+    sensorDataAtom.updateSensor.dispatch('accelerometer', {
+      x: lowerTheNumber(accelerometer.x),
+      y: lowerTheNumber(accelerometer.y),
+      z: lowerTheNumber(accelerometer.z),
+      timestamp: accelerometer.timestamp,
+    });
+  };
+  accelerometer.onerror = getOnErrorFunction(stopRecording, 'accelerometer');
+
+  // Describe orientationSensor
+  orientationSensor.onreading = () => {
+    const quaternion = orientationSensor.quaternion || [];
+    sensorDataAtom.updateSensor.dispatch('orientation', {
+      x: lowerTheNumber(quaternion[0]),
+      y: lowerTheNumber(quaternion[1]),
+      z: lowerTheNumber(quaternion[2]),
+      w: lowerTheNumber(quaternion[3]),
+      timestamp: orientationSensor.timestamp,
+    });
+  };
+  orientationSensor.onerror = getOnErrorFunction(stopRecording, 'orientation');
+
+  // Describe gyroscope
+  gyroscope.onreading = () => {
+    sensorDataAtom.updateSensor.dispatch('gyroscope', {
+      x: lowerTheNumber(gyroscope.x),
+      y: lowerTheNumber(gyroscope.y),
+      z: lowerTheNumber(gyroscope.z),
+      timestamp: gyroscope.timestamp,
+    });
+  };
+  gyroscope.onerror = getOnErrorFunction(undefined, 'gyroscope');
+}
+
+export function hookGeolocation(
+  sensorDataAtom: SensorDataAtomExportType,
+  stopRecording: () => void,
+  geolocation: Geolocation,
+  requestAction: TriggerRequestActionType,
+) {
+  // It calls prompt window to allow sharing location for the first time
+  const watchId = geolocation.watchPosition((pos) => {
+    // This function runs each second after user allowed sharing navigation
+    // Run updating sensor first
+    sensorDataAtom.updateSensor.dispatch('coordinates', {
+      lng: pos.coords.longitude,
+      lat: pos.coords.longitude,
+      alt: pos.coords.altitude,
+      accuracy: pos.coords.accuracy,
+      speed: pos.coords.speed,
+      heading: pos.coords.heading,
+      timestamp: pos.timestamp,
```

Comment:
for investigations please report the timestamp in pos object and the current time on the system clock. in case of GPS they are expected to be different, in case of wifi - very close, which might help filter out some updates later. cc @frolui

Reply:
added, not resolving in case @frolui didn't seen it yet